sdks/core/src/main/java/org/gbif/pipelines/core/converters/OccurrenceJsonConverter.java

```diff
@@ -200,10 +204,42 @@ private void mapBasicRecord(OccurrenceJsonRecord.Builder builder) {
       JsonConverter.convertVocabularyConcept(gx.getLatestAgeOrHighestStage())
           .ifPresent(gcb::setLatestAgeOrHighestStage);
 
+      GeologicalRange.Builder rangeBuilder = GeologicalRange.newBuilder();
+      chooseFirst(
+              GeologicalContextInterpreter.START_AGE_TAG,
+              gx.getEarliestAgeOrLowestStage(),
+              gx.getEarliestEpochOrLowestSeries(),
+              gx.getEarliestPeriodOrLowestSystem(),
+              gx.getEarliestEraOrLowestErathem(),
+              gx.getEarliestEonOrLowestEonothem())
+          .ifPresent(rangeBuilder::setLte);
+      chooseFirst(
+              GeologicalContextInterpreter.END_AGE_TAG,
+              gx.getLatestAgeOrHighestStage(),
+              gx.getLatestEpochOrHighestSeries(),
+              gx.getLatestPeriodOrHighestSystem(),
+              gx.getLatestEraOrHighestErathem(),
+              gx.getLatestEonOrHighestEonothem())
+          .ifPresent(rangeBuilder::setGt);
+      gcb.setRange(rangeBuilder.build());
+
       builder.setGeologicalContext(gcb.build());
     }
   }
 
+  private Optional<String> chooseFirst(String tag, VocabularyConcept... vocabularyConcepts) {
```

Comment:
It looks a bit like interpretation part, so maybe add 2 fileds to GeologicalContext avro and fill range in GeologicalContextInterpreter

Reply:
makes sense